rts/lra/lra-coordinator-jar/src/test/java/io/narayana/lra/coordinator/LRACoordinatorTestCase.java

```diff
@@ -0,0 +1,259 @@
+        try (Response response = ClientBuilder.newClient().target(lraListenerURL).path(LRA_LISTENER_ACTION)
+                .request()
+                .put(null)) {
+
+            Assert.assertEquals("LRA participant action", 200, response.getStatus());
+
+            lraId = response.readEntity(String.class);
+            fail(testName + ": byteman should have killed the container");
+        } catch (RuntimeException e) {
+            LRALogger.logger.infof("%s: byteman killed the container", testName);
+            // we could have started the LRA via lraClient (which we do in the next test) but it is useful to test the filters
+            lraId = getFirstLRA();
+            assertNotNull("LRA should have been added to the object store before byteman killed the JVM", lraId);
+            lraId = String.format("%s/%s", getCoordinatorUrl(), lraId);
+        }
+
+        // the byteman script should have killed the JVM
+        // wait for a period longer than the timeout before restarting the coordinator
+        doWait(LRA_SHORT_TIMELIMIT * 1000);
+
+        restartContainer();
+
+        // check recovery
+        LRAStatus status = getStatus(new URI(lraId));
+
+        LRALogger.logger.infof("%s: Status after restart is %s%n", status == null ? "GONE" : status.name());
+
+        if (status == null || status == LRAStatus.Cancelling) {
+            int sc = recover();
+
+            if (sc != 0) {
+                recover();
+            }
+        }
+
+        // the LRA with the short timeout should have timed out and cancelled
+        status = getStatus(new URI(lraId));
+
+        Assert.assertTrue("LRA with short timeout should have cancelled",
+                status == null || status == LRAStatus.Cancelled);
+
+        // verify that the resource was notified that the LRA finished
+        String listenerStatus = getStatusFromListener();
+
+        assertEquals("LRA listener should have been told that the final state of the LRA was cancelled",
+                LRAStatus.Cancelled.name(), listenerStatus);
+    }
+
+    /**
+     * Test that an LRA which times out while there is no running coordinator is cancelled
+     * when a coordinator is restarted.
+     *
+     * Test that an LRA which times out after a coordinator is restarted after a crash is still active
+     * @throws URISyntaxException if the LRA or revovery URIs are invalid (should never happen)
+     */
+    @Test
+    public void testRecovery2() throws URISyntaxException {
+        startContainer(null);
+
+        // start an LRA with a long timeout to validate that timed LRAs do not finish early during recovery
+        URI longLRA = lraClient.startLRA(null, "Long Timeout Recovery Test", LONG_TIMEOUT, ChronoUnit.MILLIS);
+        // start an LRA with a short timeout to validate that timed LRAs that time out when the coordinator is unavailable are cancelled
+        URI shortLRA = lraClient.startLRA(null, "Long Timeout Recovery Test", SHORT_TIMEOUT, ChronoUnit.MILLIS);
```

Comment:
different clientId can help with debugging in future

Reply:
+1